### ISSUE TITLE
New epoxy extension for BMC passwords, and port 22 for SSH

### ISSUE THREAD

**Issue by nkinkade** [m-lab/epoxy#94]
We are very close to having complete DRAC/BMC configuration happen automatically during stage1 boot. There are two things currently preventing us from deprecating the configure_drac.py script, which is still a manual step for new sites/machines:

* DRACs currently are configured to listen for SSH connections on port 806. If we left the default (port 22), then that would be one less custom thing we would need, requiring a manual step by an operator.
* stage1 boots currently give the DRACs a predefined, common and relatively insecure password. It is currently necessary to run the DRAC script afterward to configure a more secure, random password. The passwords are currently stored in Cloud Datastore. We should write a small ePoxy extension that allows the stage1 boot process to submit a random password to be stored in GCD.

**Comment by nkinkade** [comment]
Once m-lab/epoxy-extensions#1 is merged, there will still be a number of other steps to complete this work. These steps will need to be completed, possibly others

* Automated Docker Hub builds need to be set up for the extension.
* A new bmc-store-password.service systemd unit file will need to be added to [cloud-config_master.yaml in the k8s-support repo](https://github.com/m-lab/siteinfo/pull/162).
* Since cloud-config_master.yaml is only evaluated when a VM is first instantiated, the unit files will need to be manually set up on existing master nodes.
* Add a new kernel parameter with a valid URL for the extension. `epoxy.bmc_store_password`?
* The [shell script that configures DRACs during stage1 boot](https://github.com/m-lab/epoxy-images/blob/master/configs/stage1_minimal/rc.local#L56) will need to be updated to make calls to the new ePoxy extension.

**Comment by stephen-soltesz** [comment]
I like the idea of an epoxy-extensions repo.

Regarding adding and enabling the extension to reach the kernel parameters:
* Add new entry here - https://github.com/m-lab/epoxy/blob/master/storage/extensions.go#L40
* Add the new extension to the `Extensions` field in epoxy Host records in GCD - https://github.com/m-lab/epoxy/blob/master/storage/host.go#L98
* Ensure that the epoxy-images/action scripts copy the extension url through to the right stage - e.g. https://github.com/m-lab/epoxy-images/blob/master/actions/stage2/stage1to2.ipxe#L73

**Comment by nkinkade** [comment]
An additional step:
* Update the bootstrap_platform_cluster.sh script to set up the platform cluster VPC network firewall to allow port 8801 from the ePoxy subnet.

**Comment by nkinkade** [comment]
A somewhat ordered checklist of things to be done:
* [x] Build extension
* [x] Configure automated Docker Hub builds
* [ ] Update the bootstrap_platform_cluster.sh script to set up the platform cluster VPC network firewall to allow port 8801 from the ePoxy subnet.
* [ ] A new bmc-store-password.service systemd unit file will need to be added to cloud-config_master.yaml in the k8s-support repo.
* [ ] Since cloud-config_master.yaml is only evaluated when a VM is first instantiated, the unit files will need to be manually set up on existing master nodes.
* [ ] The shell script that configures DRACs during stage1 boot will need to be updated to make calls to the new ePoxy extension.
* [ ] Add a new kernel parameter with a valid URL for the extension.
* [ ] Tag epoxy-images to build new stage1 ISOs

**Comment by nkinkade** [comment]
I believe this work should now be done, but we have still not seen it work e2e in production. BOG04 will be coming up today, and I will see whether things work as expected for this new site. If so, then I'll close this issue. If not, then I'll leave it open and work to address whatever prevented things from WAI.

**Comment by nkinkade** [comment]
Turns out I had missed one important step in the process:

* [x] Manually deploy all bmc-store-password resources to staging and production.

These resources were automatically deployed to sandbox, since I baked the configs in the cluster bootstrap scripts and redeployed the cluster there. But since I didn't want to redeploy the staging and production clusters I manually deployed the resources in those projects.

mlab{1,2,3,4}-bog04 all booted, configured network, and configured admin user, and stored credentials in GCD. I believe that this is now fully complete. \o/
